### PR TITLE
retail mixin datatypes addr commerce order person prodlistitem webinf…

### PR DESCRIPTION
**Pull request by nishithshah7**
… webp prof-pers errordet

Please link to the issue #…

### CODE REVIEW

File: components/datatypes/commerce.schema.json [hunk 1]
```diff
@@ -62,6 +62,114 @@
...
+        },
+        "xdm:saveForLaterView ": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description": "Tracks when a visitor removes a product or item to their save for later list."
+        },
+        "xdm:saveForLaterRemove": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description": "Tracks when a visitor views their save for later list."
+        },
+        "xdm:checkout": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:90]: We define nested properties as
"xdm:checkout" {
 "type": "object",
  "properties": {
       All the nested properties are defined here.
  }
}

File: components/datatypes/commerce.schema.json [hunk 2]
```diff
@@ -62,6 +62,114 @@
...
+        },
+        "xdm:productListTotalValue": {
+          "$ref": "https://ns.adobe.com/xdm/data/currency",
+          "description": "Count of total value of items currently in the shopping cart. "
+        },
+        "xdm:productListPageView": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description": "Counted when a category or product listing page has been viewed."
+        },
+        "xdm:wishList": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:128]: please refer to the checkout property comment

File: components/datatypes/commerce.schema.json [hunk 3]
```diff
@@ -62,6 +62,114 @@
...
+          "xdm:wishListView": {
+            "$ref": "https://ns.adobe.com/xdm/data/measure",
+            "description": "Tracks when a visitor views a product or item to their wish list."
+          },
+          "xdm:wishListSave": {
+            "$ref": "https://ns.adobe.com/xdm/data/measure",
+            "description": "Tracks when a visitor saves a product or item to their wish list."
+          }
+        },
+        "xdm:productReviewed": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:142]: see prev comments

File: components/datatypes/commerce.schema.json [hunk 4]
```diff
@@ -62,6 +62,114 @@
...
+          "xdm:productId": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:143]: All id are defined as ID (productID). Please apply everywhere

File: components/datatypes/data/order.schema.json [hunk 5]
```diff
@@ -42,6 +42,125 @@
           "title": "Price Total",
           "type": "number",
           "description": "The total price of this order after all discounts and taxes have been applied."
+        },
+        "xdm:promoCode": {
+          "title": "Promotional Code",
+          "type": "array",
```
Comment (mjasrotia) [inline components/datatypes/data/order.schema.json:48]: Please see an existing example to define arrays

File: components/datatypes/data/order.schema.json [hunk 6]
```diff
@@ -42,6 +42,126 @@
...
+          "title": "Store ID",
+          "type": "string",
+          "description": "Store ID of BOPIS or BOSS store."
+        },
+        "xdm:deliveryType": {
+          "title": "Order Delivery Type",
+          "type": "string",
+          "description": "Order Delivery Type."
+        },
+        "xdm:shipToHome": {
```
Comment (mjasrotia) [inline components/datatypes/data/order.schema.json:142]: Should this be a Boolean ?

File: components/datatypes/commerce.schema.json [hunk 7]
```diff
@@ -23,10 +23,6 @@
           "$ref": "https://ns.adobe.com/xdm/data/measure",
           "description": "A product list has been identified as no longer accessible or purchasable by the user."
         },
-        "xdm:checkouts": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:26]: Can't remove properties like that. Please deprecate by adding "meta:status": "deprecated". 
`
        "xdm:checkouts": {
          "$ref": "https://ns.adobe.com/xdm/data/measure",
          "description": "An action during a checkout process of a product list, there can be more than one checkout event if there are multiple steps in a checkout process. If there are multiple steps the event time information and referenced page or experience is used to identify the step individual events represent in order.",
        "meta:status": "deprecated",
        },
`

File: components/datatypes/commerce.schema.json [hunk 8]
```diff
@@ -23,10 +23,6 @@
...
-          "$ref": "https://ns.adobe.com/xdm/data/measure",
-          "description": "An action during a checkout process of a product list, there can be more than one checkout event if there are multiple steps in a checkout process. If there are multiple steps the event time information and referenced page or experience is used to identify the step individual events represent in order."
-        },
         "xdm:productListAdds": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:26]: Every property needs to have a "title". A lot of properties in this PR are missing one, Please add that.

File: components/datatypes/commerce.schema.json [hunk 9]
```diff
@@ -62,6 +58,133 @@
...
+              "title": "Product Review Rating",
+              "description": "Tracks the rating of a product review.",
+              "type": "string"
+            },
+            "xdm:reviewBody": {
+              "title": "Product Review Comments",
+              "description": "Tracks the text content of a product review.",
+              "type": "string"
+            },
+            "xdm:reviewId": {
```
Comment (mjasrotia) [inline components/datatypes/commerce.schema.json:171]: Id needs to be ID. Please check other IDs too

File: components/datatypes/data/order.schema.json [hunk 10]
```diff
@@ -42,6 +42,126 @@
...
+          "title": "Order Ship to Home",
+          "type": "string",
+          "description": "Ship to home? Yes/No."
+        },
+        "xdm:saveBillingInformation": {
+          "title": "Save Billing Information",
+          "type": "string",
+          "description": "Tracks if a visitor chooses to have their billing information stored for future orders."
+        },
+        "xdm:splitPaymentMethods": {
```
Comment (mjasrotia) [inline components/datatypes/data/order.schema.json:152]: Should this be an array ?

File: components/datatypes/productlistitem.schema.json [hunk 11]
```diff
@@ -56,6 +56,77 @@
...
+          "title": "Product SKU Variant",
+          "type": "string",
+          "description": "Tracks if a product has alternate SKUs or IDs, such as different SKUs for color variants."
+        },
+        "xdm:brand": {
+          "title": "Product Brand",
+          "type": "string",
+          "description": "Tracks the brand of the individual product."
+        },
+        "xdm:category": {
```
Comment (mjasrotia) [inline components/datatypes/productlistitem.schema.json:81]: Can you please get in touch with Rachel/Prabhakar to see if you can use the standard product classes/mixins to populate category and sub-category. They are coming up with a bunch of Product mixins.

File: components/datatypes/productlistitem.schema.json [hunk 12]
```diff
@@ -56,6 +56,77 @@
...
+          "title": "Product Subcategory",
+          "type": "string",
+          "description": "Tracks the child/sub category for the individual product."
+        },
+        "xdm:inventory": {
+          "title": "Product Inventory Count",
+          "type": "integer",
+          "description": "Tracks the current inventory for the individual product."
+        },
+        "xdm:productRating": {
```
Comment (mjasrotia) [inline components/datatypes/productlistitem.schema.json:96]: Could a rating be in decimals like 4.5 ? If yes, please convert this to number

File: components/datatypes/webpagedetails.schema.json [hunk 13]
```diff
@@ -54,6 +54,48 @@
           "title": "Page Views",
           "$ref": "https://ns.adobe.com/xdm/data/measure",
           "description": "View(s) of a webpage has occurred."
+        },
+        "xdm:primarySiteSection": {
+          "title": "Primary Site Section",
+          "type": "string",
+          "description": "The first level/section of the site hierarchy."
+        },
+        "xdm:primarySiteSectionId": {
```
Comment (mjasrotia) [inline components/datatypes/webpagedetails.schema.json:63]: ID

**Comment by nishithshah7** [comment]
Will open PR for individual mixins

**Pull request closed without merging**